hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/query/AdjacentEdgesQuery.java

```diff
@@ -0,0 +1,396 @@
+                return null;
+            }
+            if (this.edgeLabels.length == 1) {
+                return Condition.eq(HugeKeys.LABEL, this.edgeLabels[0]);
+            }
+            return Condition.in(HugeKeys.LABEL, Arrays.asList(this.edgeLabels));
+        }
+        return null;
+    }
+
+    @Override
+    public boolean containsCondition(HugeKeys key) {
+        if (key == HugeKeys.OWNER_VERTEX) {
+            return true;
+        }
+        if (key == HugeKeys.DIRECTION) {
+            return true;
+        }
+        if (key == HugeKeys.LABEL) {
+            if (this.edgeLabels.length == 0) {
+                return false;
+            }
+            return true;
+        }
+        return super.containsCondition(key);
+    }
+
+    @Override
+    public void unsetCondition(Object key) {
+        if (key == HugeKeys.OWNER_VERTEX ||
+            key == HugeKeys.DIRECTION ||
+            key == HugeKeys.LABEL) {
+            E.checkArgument(false, "Can't unset condition '%s'", key);
+        }
+        super.unsetCondition(key);
+    }
+
+    @Override
+    public List<Condition> syspropConditions() {
+        List<Condition> conds = this.selfConditions();
+        if (super.conditionsSize() > 0) {
+            conds.addAll(super.syspropConditions());
+        }
+        return conds;
+    }
+
+    @Override
+    public List<Condition> syspropConditions(HugeKeys key) {
+        Condition cond = this.selfRelation(key);
+        if (cond != null) {
+            return ImmutableList.of(cond);
+        }
+        return super.syspropConditions(key);
+    }
+
+    @Override
+    public List<Condition.Relation> relations() {
+        // NOTE: selfConditions() actually return a list of Relation
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        List<Condition.Relation> relations = (List) this.selfConditions();
+        if (super.conditionsSize() > 0) {
+            relations.addAll(super.relations());
+        }
+        return relations;
+    }
+
+    @Override
+    public Relation relation(Id key){
+        Relation relation = this.selfRelation(key);
+        if (relation != null) {
+            return relation;
+        }
+        return super.relation(key);
+    }
+
+    @Override
+    public boolean allSysprop() {
+        return super.allSysprop();
+    }
+
+    @Override
+    public boolean allRelation() {
+        if (!this.isFlattened()) {
+            return false;
+        }
+        return super.allRelation();
+    }
+
+    @Override
+    public AdjacentEdgesQuery copy() {
+        return (AdjacentEdgesQuery) super.copy();
+    }
+
+    @Override
+    public AdjacentEdgesQuery copyAndResetUnshared() {
+        return (AdjacentEdgesQuery) super.copyAndResetUnshared();
+    }
+
+    @Override
+    public boolean isFlattened() {
+        if (this.direction == Directions.BOTH) {
+            return false;
+        }
+        if (this.edgeLabels.length > 1) {
+            return false;
+        }
```

Comment:
Consider using Collections.emptyList() instead of creating a new ArrayList when no flattening is needed for better memory efficiency.